### PR TITLE
Issue 5339 unable to upload ios build to appstore

### LINKED ISSUE #5339: Unable to upload iOS builds to the App Store
**Describe the bug (REQUIRED)**
A few problems when uploading to the app store:

* iPad multitasking requires multiple orientations specified in Info.plist. Can be solved by adding `UIRequiresFullScreen`.
* The Assets.car file is copied with folders. It must always exist in the root of the app.
* The instructions for creating the Assets.car file were not specific enough. The Xcode project must build with the "Any iOS Device (arm64)" schema to generate an Assets.car with all icons.
* Icon names in Info.plist should match the ones Xcode uses. This will help with debugging and validation against an Xcode generated project.

**To Reproduce (REQUIRED)**
1. Bundle for iOS
2. Upload using the Transporter tool
3. Notice errors

**Expected behavior (REQUIRED)**
Application uploads should work without errors.

**Defold version (REQUIRED):**
  - Version 1.2.175

**Platforms (REQUIRED):**
 - Platforms: iOS

### PR DESCRIPTION
**Pull request by britzl**

### CODE REVIEW

**Comment by britzl** [comment]
We should cherry-pick and merge with the 1.2.176 bugfix beta once this is approved

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/BundleHelper.java [hunk 1]
```diff
@@ -608,11 +608,11 @@ public void copyIosIcons() throws IOException
         BufferedImage largestIconImage = getFallbackIconImage("ios", iconPropNames);
 
         // Copy game.project specified icons
-        genIcon(largestIconImage, appDir, "ios",   "app_icon_76x76",      "Icon-76.png",  76);
-        genIcon(largestIconImage, appDir, "ios", "app_icon_152x152",   "Icon-76@2x.png", 152);
-        genIcon(largestIconImage, appDir, "ios", "app_icon_120x120",   "Icon-60@2x.png", 120);
-        genIcon(largestIconImage, appDir, "ios", "app_icon_180x180",   "Icon-60@3x.png", 180);
-        genIcon(largestIconImage, appDir, "ios", "app_icon_167x167", "Icon-83.5@2x.png", 167);
+        genIcon(largestIconImage, appDir, "ios", "app_icon_120x120",          "AppIcon60x60@2x.png", 120);
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/BundleHelper.java:611]: Renaming these to be the same as in a default Xcode project. This makes it easier to compare a Defold bundle with and Xcode bundle when searching for discrepancies.

File: com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java [hunk 2]
```diff
@@ -269,13 +269,13 @@ public void bundleApplication(Project project, File bundleDir, ICanceled cancele
...
         {
-            final String iconsBaseName = FilenameUtils.getName(iconsAsset);
+            final String iconsName = FilenameUtils.getName(iconsAsset);
             IResource source = project.getResource(iconsAsset);
             if (source == null) {
                 throw new IOException(String.format("'ios.icons_asset' = '%s' does not exist", source));
             }
 
-            File target = new File(appDir, iconsAsset);
+            File target = new File(appDir, iconsName);
```
Comment (britzl) [inline com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/bundle/IOSBundler.java:278]: Always copy the Assets.car to the root of the project

File: engine/engine/content/builtins/manifests/ios/Info.plist [hunk 3]
```diff
@@ -18,9 +18,22 @@
...
-                                <string>Icon-60</string>
-                                <string>Icon-76</string>
-                                <string>Icon-83.5</string>
+                                <string>AppIcon60x60</string>
+                        </array>
+                        <key>UIPrerenderedIcon</key>
+                        <{{ios.pre_renderered_icons}}/>
+                </dict>
+        </dict>
+        <key>CFBundleIcons~ipad</key>
```
Comment (britzl) [inline engine/engine/content/builtins/manifests/ios/Info.plist:27]: Split into iphone and ipad app icons

File: engine/engine/content/builtins/manifests/ios/Info.plist [hunk 4]
```diff
@@ -104,5 +117,8 @@
 
         <key>UILaunchStoryboardName</key>
         <string>{{launch-screen}}</string>
+
+        <key>UIRequiresFullScreen</key>
```
Comment (britzl) [inline engine/engine/content/builtins/manifests/ios/Info.plist:121]: Required to avoid the iPad multitasking error when uploading

**Review by JCash — APPROVED** [review]
Looks good

**Pull request merged**